src/main/api/2/studio-api-2.yaml

```diff
@@ -4230,6 +4230,56 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /content/get_content:
+    get:
+      tags:
+        - content
+      summary: Get content for a path
+      description: 'Required permission "content_read" and additional "content_write" if edit parameter is `true`'
+      operationId: getContent
+      parameters:
+        - name: siteId
+          in: query
+          description: site ID
+          required: true
+          schema:
+            type: string
+        - name: path
+          in: query
+          description: path of the content
+          required: true
+          schema:
+            type: string
+        - name: edit
```

Comment:
What this really means is to lock the content so it is not edited by someone else.
Should we explicitly call this `lock` / `lockContent`?
@sumerjabri @phuongnq

Reply:
We will create a small design doc to decide:
* API name
* Params and expectations

I think we should expect this one to work for any item (binary or text), and for locking to happen externally (lock, readItem, unlock).

More to come via design doc.

Reply:
@phuongnq I have shared a doc with you. It is still in progress, but you can refer to the section about this API.

"Get content" is pretty much already implemented by "get_content_by_commit_id". Get content will rely on that functionality using HEAD as the commitId.
Required changes:
- Create new mapping in ContentController for a request like  "/content/{siteId}?path=abc&commitId=123". CommitId is optional, default to `HEAD`
- Keep "/get_content_by_commit_id" mapping, but mark it as deprecated.
- You might need to create a private method to perform the actual service invoking, since the mappings mix path and request parameters.
- Update studio-api-2.yaml accordingly. Include a note for deprecated URL.

Please let me know if you have any questions/suggestions/etc.


cc @sumerjabri

Reply:
I updated the API as per the above discussion. Thank you!